### ISSUE TITLE
Filter search engines gives no results

### ISSUE THREAD

**Issue by flozi00** [searx/searx#1933]
Trying to filter the search engines using engines=bing or engines=duckduckgo
Everytime there is no result
Using multiple public instances and own instances

Only engines=google is working

**Comment by unixfox** [comment]
Duplicate of https://github.com/asciimoo/searx/issues/1496

**Comment by unixfox** [comment]
Public Searx instances can get blocked by various search engines pretty quickly.

For DuckDuckGo, it appears recently to block IPs that do an unusual amount of requests to their servers. After a few requests it returns a 403 HTTP status then block the IP entirely.
And about Bing, it seems to do to the same behavior but it silently fails with returning no results. Changing the user agent to a newer one seems to really help with the block, I made a [PR](https://github.com/asciimoo/searx/pull/1934) for that. Funny enough it returns only the results when using Firefox 75 as a user agent but not for Firefox 74 (gotta love Microsoft).

If you would like to get a better experience with Searx when using DDG or Bing you should run your own private Searx instance.
But if for some reason you can't, you can use my instance which has a "rotating proxy" which essentially tries to unblock the instance from DDG and Bing: https://searx.be.

I hope this help with your issue.

**Comment by flozi00** [comment]
Thank you very much
How did you implement the rotating proxy ?

**Comment by return42** [comment]
Very simple, by setting the header https://github.com/asciimoo/searx/blob/master/searx/search.py#L475 .. by randomly `choice` https://github.com/asciimoo/searx/blob/master/searx/utils.py#L65

Thanks for your issue, I will now close it since the problem is solved .. right?

**Comment by flozi00** [comment]
what about really rotating proxy using public proxy ip lists ?
maybe a cronjob to check if proxy works and generate own working list to use ?
example: https://github.com/danilopolani/rotating-proxy-python/blob/master/scrape_me.py

**Comment by return42** [comment]
Sorry, you asked about proxy and I answered about the HTTP header .. I think I need a coffee first, before answering questions ...

**Comment by unixfox** [comment]
I had to modify some code from the source code of Searx for that but it's very specific for my use case so I won't dig into the details. But another way to achieve the same result is to create a separate proxy that rotate the IP using public proxies and plug that into searx by modifying the proxy section in the `settings.yml` file.
All of these modifications are not needed for a private searx instance though.

**Comment by return42** [comment]
FYI: further discussion about proxy can be done here https://github.com/asciimoo/searx/issues/1512

**Comment by andrew6838** [comment]
> Public Searx instances can get blocked by various search engines pretty quickly.
> 
> For DuckDuckGo, it appears recently to block IPs that do an unusual amount of requests to their servers. After a few requests it returns a 403 HTTP status then block the IP entirely.
> And about Bing, it seems to do to the same behavior but it silently fails with returning no results. Changing the user agent to a newer one seems to really help with the block, I made a [PR](https://github.com/asciimoo/searx/pull/1934) for that. Funny enough it returns only the results when using Firefox 75 as a user agent but not for Firefox 74 (gotta love Microsoft).
> 
> If you would like to get a better experience with Searx when using DDG or Bing you should run your own private Searx instance.
> But if for some reason you can't, you can use my instance which has a "rotating proxy" which essentially tries to unblock the instance from DDG and Bing: https://searx.be.
> 
> I hope this help with your issue.

Searx.be is working good for google result, my one is not.
Can you share how you make?

I still always get this error. Sigh.
google (unexpected crash CAPTCHA required)

**Comment by unixfox** [comment]
> Can you share how you make?

I replied here: https://github.com/searx/searx/issues/2161#issuecomment-686945342

> I still always get this error. Sigh.
> google (unexpected crash CAPTCHA required)
> 

Do you mind asking your question in the specific issue: https://github.com/searx/searx/issues/729? We can talk about it there.
